lib/runner.go

```diff
@@ -97,14 +99,17 @@ type MiniRunner struct {
 }
 
 func (r MiniRunner) VU(out chan<- stats.SampleContainer) *MiniRunnerVU {
-	return &MiniRunnerVU{R: r, Out: out}
+	return &MiniRunnerVU{R: r, Out: out, ID: rand.Int63()}
 }
 
 func (r MiniRunner) MakeArchive() *Archive {
 	return nil
 }
 
 func (r MiniRunner) NewVU(out chan<- stats.SampleContainer) (VU, error) {
+	// XXX: This method isn't called by the new executors.
+	// Confirm whether this was intentional and if other methods of
+	// the Runner interface are unused.
```

Comment:
@na-- Can you comment on this?

Is calling `ExecutionState.SetInitVUFunc()` required? I would expect `Runner.NewVU()` to be called for this purpose instead.

Reply:
ping @na-- :)

Reply:
Sorry, this was buried in the mountain of emails, saw it just after you merged... :disappointed: Not sure why you think this isn't used:
https://github.com/loadimpact/k6/blob/eda42f568cda135300adcf8a25e38ae937ec5899/core/local/local.go#L240-L242
https://github.com/loadimpact/k6/blob/eda42f568cda135300adcf8a25e38ae937ec5899/core/local/local.go#L135-L138